controllers/clusterversion_controller.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2023 Red Hat OpenShift Data Foundation.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/red-hat-storage/ocs-client-operator/pkg/csi"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/templates"
+
+	configv1 "github.com/openshift/api/config/v1"
+	secv1 "github.com/openshift/api/security/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// ClusterVersionReconciler reconciles a ClusterVersion object
+type ClusterVersionReconciler struct {
+	client.Client
+	OperatorDeployment *appsv1.Deployment
+	OperatorNamespace  string
+	ctx                context.Context
+	Log                klog.Logger
+	Scheme             *runtime.Scheme
+
+	cephFSDeployment *appsv1.Deployment
+	cephFSDaemonSet  *appsv1.DaemonSet
+	rbdDeployment    *appsv1.Deployment
+	rbdDaemonSet     *appsv1.DaemonSet
+	scc              *secv1.SecurityContextConstraints
+}
```

Comment:
Is there a general pattern for which ones are capitalized and which ones aren't? I imagine the reconciler struct in general isn't meant to be used by anything other than itself, so I'd think they'd be all lowercase. I'm not hardline one way or the other, but a consistent rationale would help things down the line.

Reply:
The values that need to be set in  the main.go are exported all others are unexported.